pkg/comp-functions/functions/vshnpostgres/postgresql_deploy.go

```diff
@@ -394,6 +394,27 @@ func createSgCluster(ctx context.Context, comp *vshnv1.VSHNPostgreSQL, svc *runt
 
 	l := svc.Log
 
+	// Determine whether this is a new instance by checking for the existence of SGCluster
+	isNewInstance := common.IsNewInstance(svc, &sgv1.SGCluster{}, "cluster")
+
+	disableEnvoy := false
```

Comment:
You could make this a `*bool` then you won't have to use `*` and `ptr.To` later on.

Reply:
we could simplify the code below by keeping the type `bool`.  I would avoid a third value such as `nil` for this. And at this line of code you could just assign the value that you already have in the composite.